src/kernel.h

```diff
@@ -154,7 +172,25 @@ class quantum_kernel {
         const utils::Vec<utils::UInt> &qubits = {},
         const utils::Vec<utils::UInt> &cregs = {},
         utils::UInt duration = 0,
-        utils::Real angle = 0.0
+        utils::Real angle = 0.0,
+        const utils::Vec<utils::UInt> &bregs = {},
+        cond_type_t gcond = cond_always,
+        const utils::Vec<utils::UInt> &gcondregs = {}
+    );
+    void gate_preset_condition(
+        cond_type_t gcond,
+        const utils::Vec<utils::UInt> &gcondregs
+    );
+    void gate_clear_condition();
+    void gate_add_implicits(
```

Comment:
Should probably be private

Reply:
gate_add_implicits must be private, the clear_condition not since it is an interface; see above.